numba/targets/arrayobj.py

```diff
@@ -188,7 +187,9 @@ def getiter_array(context, builder, sig, args):
     iterobj.index = indexptr
     iterobj.array = array
 
-    return iterobj._getvalue()
+    res = iterobj._getvalue()
+    out = impl_ret_borrowed(context, builder, sig.return_type, res)
```

Comment:
I am taking advantage of the fact that ``context.nrt_incref`` traverse the structure to incref all ``MemInfo*`` (same for ``nrt_decref``).  I will change this to make it reflect what it is actually doing.  But, note that the decref will be applying on the ``iterobj``